### PR TITLE
feature/DatetimeSinusoidCalculator

### PR DESCRIPTION
**Pull request by davidhopkinson26**
A transformer that calculates sin or cos of any time unit.

### CODE REVIEW

File: tubular/dates.py [hunk 1]
```diff
@@ -663,3 +665,152 @@ def transform(self, X):
...
+            raise TypeError(
+                "{}: units must be a string but got {}".format(
+                    self.classname(), type(units)
+                )
+            )
+
+        if (
+            (not isinstance(period, int))
+            and (not isinstance(period, float))
+            or isinstance(period, bool)
```
Comment (TommyMatthews) [inline tubular/dates.py:736]: Think the bool option here is left over from when I had the option to set `period=False`

File: tubular/dates.py [hunk 2]
```diff
@@ -663,3 +665,152 @@ def transform(self, X):
         X[self.new_column_name] = lower_comparison & upper_comparison
 
         return X
+
+
+class DatetimeSinusoidCalculator(BaseTransformer):
+
+    """
+    Transformer to derive a feature in a dataframe by calculating the
```
Comment (TommyMatthews) [inline tubular/dates.py:673]: Looking at this transformer again, I think it seems a bit inconsistent that you can specify multiple input columns but only one unit or period - might be a bit much for now but could be worth adding a refactor to the backlog (e.g. to dictionaries or lists that are assumed to have matching inputs)

File: tests/dates/test_DatetimeSinusoidCalculator.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+                "columns",
+                "method",
+                "units",
+                "period",
+            ],
+            expected_default_values=(2 * np.pi,),
+        )
+
+    def test_class_methods(self, example_transformer):
+        """Test that DateDifferenceTransformer has a transform methods."""
```
Comment (TommyMatthews) [inline tests/dates/test_DatetimeSinusoidCalculator.py:37]: method

Comment (davidhopkinson26) [reply tests/dates/test_DatetimeSinusoidCalculator.py]: https://github.com/lvgig/tubular/pull/50/commits/eff8469d47e048a8634eb8227f5491c0a048c531

File: tubular/dates.py (hunk 1 above)
Comment (davidhopkinson26) [reply tubular/dates.py]: removed this line and relevant test case: 
https://github.com/lvgig/tubular/pull/50/commits/eff8469d47e048a8634eb8227f5491c0a048c531

**Review by TommyMatthews — APPROVED** [review]

**Pull request merged**